forge/csrc/autograd/autograd.cpp

```diff
@@ -719,6 +721,152 @@ NodeContext autograd_engine::create_input(
     return NodeContext(node);
 }
 
+static Node *create_recompute_op(
```

Comment:
Can we add a brief doc regarding how this function works? E.g., focusing on mapping fwd node(s) and replacing specific bwd node.